### DISCUSSION TITLE
Extend all Arg types to include the same fields and omit unneeded fields with type never

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#22183] [Q&A]
### Proposition
It would be great to extend all Argument types to include the same fields. Currently when we try to access the utility type Args<Type, Operation>[FieldName] FieldName can only be common to each operation arguments. For example for create operation we have only 'data' and 'select', however indexing 'include' is not possible. If each function argument includes the same fields (where we do not want a field we can impose fiels type equals to never) indexing 'include' would be possible.

### How to reproduce (optional)

1. Create Prisma schema that include two related tables and one that's not related to anything.
```
model make {
  id        Int     @id @default(autoincrement())
  name      String  @unique
  model     model[]
}

model model {
  id        Int     @id @default(autoincrement())
  name      String  @unique
  makeName  String
  make      make    @relation(fields: [makeName], references: [name])
}

model fuel_type {
  id        Int     @id @default(autoincrement())
  name      String  @unique
  isDeleted Boolean @default(false)
}

```
2. Try to use utility type Args<Type, Operation>
`const a: Prisma.Args<PrismaService[M], 'create'>['include'] = {.....}`

### Expected behavior (optional)

Being able to index the type Prisma.Args by field "include" although it is not usable for model fuel_type.
Prisma.Args<PrismaService[M], 'create'>['include']

### Actual behavior (optional)

In reality it is not possible, however if fuel_type arguments include the field with type never

```
{
......
include: never
......
}
```

it would have been possible. There is a workaround by create a type like the following:
`type AddIncludeArg<T> = T extends { include?: any } ? T : T & { include?: never }`
however it is not a prefered solution to the problem. Thank you!

**Comment by ludralph** [comment]
Hi @vtonchev 👋 

Thank you for raising this question.

Your suggestion for extending Arg types to include the same fields and omitting unneeded fields with type never looks like it could be beneficial to other users as well. Do you mind creating a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that our engineering team can also provide their insights?

**Reply by ghost** [reply]
Link to feature request: https://github.com/prisma/prisma/issues/22382
